pkg/client/record/event.go

```diff
@@ -86,6 +89,9 @@ type EventBroadcaster interface {
 	// NewRecorder returns an EventRecorder that can be used to send events to this EventBroadcaster
 	// with the event source set to the given event source.
 	NewRecorder(source api.EventSource) EventRecorder
+
+	// NewRecorderWithClock makes it easy to write test with a fake clock.
+	NewRecorderWithClock(source api.EventSource, clock util.Clock) EventRecorder
```

Comment:
A less desirable way to do it (but much prefered to adding this method to the public interface) would be to type assert the interface to `*eventBroadcasterImpl` after calling `NewBroadcaster` in the tests, and then calling a private method on that.